vllm/v1/attention/backends/pallas.py

```diff
@@ -17,6 +17,9 @@
 
 logger = init_logger(__name__)
 
+# TPU requires the head size to be a multiple of 128.
```

Comment:
Is it more of a Pallas kernel requirement?

Reply:
I believe the fundamental issue lies with the TPU hardware. We need to implement padding, either within the model or the kernel. In this case, we've opted to pad at the model level.